larpandoracontent/LArHelpers/LArGeometryHelper.cc

```diff
@@ -400,6 +400,21 @@ CartesianVector LArGeometryHelper::GetWireAxis(const Pandora &pandora, const Hit
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
+std::set<unsigned int> LArGeometryHelper::GetCommonDaughterVolumes(const Cluster *const pCluster1, const Cluster *const pCluster2)
+{
+    std::set<unsigned int> intersect;
+
+    const std::set<unsigned int> daughterVolumeIds1(LArClusterHelper::GetDaughterVolumeIDs(pCluster1)), 
+        daughterVolumeIds2(LArClusterHelper::GetDaughterVolumeIDs(pCluster2));
+
+    std::set_intersection(daughterVolumeIds1.begin(), daughterVolumeIds1.end(), daughterVolumeIds2.begin(), daughterVolumeIds2.end(), 
+                          std::inserter(intersect, intersect.begin())); 
+
+    return intersect;
```

Comment:
Might be preferable to pass in the set by reference and avoid the copy constructor calls for the return value.